tests/test_transforms.py

```diff
@@ -172,18 +177,236 @@ def test_transformed_logprob(at_dist, dist_params, sp_dist, size):
+        sp.stats.norm(loc_test_val, 1).logpdf(y_test_val).sum(),
+    )
+
+
+@pytest.mark.parametrize("size", [None, 1, 2, (3, 2)])
+def test_scalar_scale_transform(size):
+    scale = at.scalar("scale")
+    y_rv = at.random.normal(0, 1, name="y_rv", size=size) * scale
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([scale, y], logp)
+
+    loc_test_val = 4.0
+    y_test_val = np.full(size, 1.0)
+
+    assert np.isclose(
+        logp_fn(loc_test_val, y_test_val),
+        sp.stats.norm(0, loc_test_val).logpdf(y_test_val).sum(),
+    )
+
+
+@pytest.mark.parametrize("size", [None, 1, 2, (3, 2)])
+def test_scalar_affine_transform(size):
+    loc = at.scalar("loc")
+    scale = at.scalar("scale")
+    y_rv = loc + at.random.normal(0, 1, name="y_rv", size=size) * scale
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([loc, scale, y], logp)
+
+    loc_test_val = 4.0
+    scale_test_val = 0.5
+    y_test_val = np.full(size, 1.0)
+
+    assert np.isclose(
+        logp_fn(loc_test_val, scale_test_val, y_test_val),
+        sp.stats.norm(loc_test_val, scale_test_val).logpdf(y_test_val).sum(),
+    )
+
+
+def test_double_scalar_loc_transform():
+    loc1 = at.scalar("loc1")
+    loc2 = at.scalar("loc2")
+    y_rv = loc1 + loc2 + at.random.normal(0, 1, name="y_rv")
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y})
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([loc1, loc2, y], logp)
+
+    loc1_test_val = 2.0
+    loc2_test_val = 2.0
+    y_test_val = 1.0
+
+    assert np.isclose(
+        logp_fn(loc1_test_val, loc1_test_val, y_test_val),
+        sp.stats.norm(loc1_test_val + loc2_test_val, 1).logpdf(y_test_val),
+    )
+
+
+def test_double_scalar_scale_transform():
+    scale1 = at.scalar("scale1")
+    scale2 = at.scalar("scale2")
+    y_rv = at.random.normal(0, 1, name="y_rv") * scale1 * scale2
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y})
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([scale1, scale2, y], logp)
+
+    scale1_test_val = 2.0
+    scale2_test_val = 2.0
+    y_test_val = 1.0
+
+    assert np.isclose(
+        logp_fn(scale1_test_val, scale1_test_val, y_test_val),
+        sp.stats.norm(0, scale1_test_val * scale2_test_val).logpdf(y_test_val),
+    )
+
+
+# TODO: Have to figure this out. Perhaps safer to not allow discrete transforms
+#  for the time being as many discrete logps are not designed for "invalid"
+#  decimal inputs
+def test_discrete_loc_transform_fails():
+    loc = at.lscalar("loc")
+    y_rv = loc + at.random.poisson(1, name="y_rv")
+    y = y_rv.type()
+    y.name = "y"
+
+    with pytest.raises(NotImplementedError):
+        joint_logprob(y_rv, {y_rv: y})
+
+
+def test_broadcasted_loc_transform_fails():
```

Comment:
Would a shape assert be a reasonable approach here? This kind of issue crops up in several of the rewrites so it's also fine for me to leave on open issue for this and move on